app/controllers/posts_controller.rb

```diff
@@ -37,4 +45,12 @@ def create
   def post_params
     params.require(:post).permit(:title, :text)
   end
+
+  def destroy
+    @post = Post.find(params[:id])
+    @author = @post.author
+    @author.decrement!(:post_counter)
+    @post.destroy!
+    redirect_to user_posts_path(id: @author.id), notice: 'The post has been successfully removed!'
+  end
```

Comment:
- [ ] Great effort working on the changes requested by the previous reviewer, but currently, I still find it difficult to delete a post and whenever I try to it throw an error. Kindly work on it and be sure everything works fine

Error: 👇 

![post](https://github.com/Marcraphael12/rails-blog-app/assets/78634841/a2e8e1d2-4966-48da-bdf1-16372875eb11)